// @prisma/internals/dist/chunk-7RMFZ2BB.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var chunk_7RMFZ2BB_exports = {};
__export(chunk_7RMFZ2BB_exports, {
  GetConfigError: () => GetConfigError,
  getConfig: () => getConfig
});
module.exports = __toCommonJS(chunk_7RMFZ2BB_exports);
var import_chunk_WSZXPLJQ = require("./chunk-WSZXPLJQ.js");
var import_chunk_R6QH57HZ = require("./chunk-R6QH57HZ.js");
var import_chunk_C47SCASR = require("./chunk-C47SCASR.js");
var import_chunk_6Y7FKZ6B = require("./chunk-6Y7FKZ6B.js");
var import_chunk_SMJRP5CD = require("./chunk-SMJRP5CD.js");
var import_chunk_5DBOS77Y = require("./chunk-5DBOS77Y.js");
var import_chunk_XKZ6CBLA = require("./chunk-XKZ6CBLA.js");
var import_chunk_LMVSIVKQ = require("./chunk-LMVSIVKQ.js");
var import_chunk_PG5FDKSF = require("./chunk-PG5FDKSF.js");
var import_chunk_4VNS5WPM = require("./chunk-4VNS5WPM.js");
var import_debug = __toESM(require("@prisma/debug"));
var import_get_platform = require("@prisma/get-platform");
var import_function = (0, import_chunk_4VNS5WPM.__toESM)((0, import_chunk_R6QH57HZ.require_function)());
var debug = (0, import_debug.default)("prisma:getConfig");
var SCHEMA_VALIDATION_ERROR_CODE = "P1012";
var GetConfigError = class extends Error {
  constructor(params) {
    const constructedErrorMessage = (0, import_chunk_XKZ6CBLA.z)(params).with({ _tag: "parsed" }, ({ errorCode, message, reason }) => {
      const errorCodeMessage = errorCode ? `Error code: ${errorCode}` : "";
      return `${reason}
${errorCodeMessage}
${message}`;
    }).with({ _tag: "unparsed" }, ({ message, reason }) => {
      const detailsHeader = (0, import_chunk_PG5FDKSF.red)((0, import_chunk_PG5FDKSF.bold)("Details:"));
      return `${reason}
${detailsHeader} ${message}`;
    }).exhaustive();
    const errorMessageWithContext = `${constructedErrorMessage}
[Context: getConfig]`;
    super((0, import_chunk_6Y7FKZ6B.addVersionDetailsToErrorMessage)(errorMessageWithContext));
    this.name = "GetConfigError";
  }
};
async function getConfig(options) {
  const debugErrorType = (0, import_chunk_WSZXPLJQ.createDebugErrorType)(debug, "getConfigWasm");
  debug(`Using getConfig Wasm`);
  const configEither = (0, import_function.pipe)(
    (0, import_chunk_5DBOS77Y.tryCatch)(
      () => {
        if (process.env.FORCE_PANIC_GET_CONFIG) {
          debug("Triggering a Rust panic...");
          import_chunk_SMJRP5CD.prismaSchemaWasm.debug_panic();
        }
        const params = JSON.stringify({ prismaSchema: options.datamodel });
        return import_chunk_SMJRP5CD.prismaSchemaWasm.get_config(params);
      },
      (e) => ({
        type: "wasm-error",
        reason: "(get-config wasm)",
        error: e
      })
    ),
    (0, import_chunk_5DBOS77Y.map)((result) => ({ result })),
    (0, import_chunk_5DBOS77Y.chainW)(
      ({ result }) => (
        // NOTE: this should never fail, as we expect returned values to be valid JSON-serializable strings
        (0, import_chunk_5DBOS77Y.tryCatch)(
          () => JSON.parse(result),
          (e) => ({
            type: "parse-json",
            reason: "Unable to parse JSON",
            error: e
          })
        )
      )
    ),
    (0, import_chunk_5DBOS77Y.chainW)((response) => {
      if (response.errors.length > 0) {
        return (0, import_chunk_5DBOS77Y.left)({
          type: "validation-error",
          reason: "(get-config wasm)",
          error: response.errors
        });
      }
      return (0, import_chunk_5DBOS77Y.right)(response.config);
    })
  );
  if ((0, import_chunk_5DBOS77Y.isRight)(configEither)) {
    debug("config data retrieved without errors in getConfig Wasm");
    const { right: data } = configEither;
    for (const generator of data.generators) {
      await resolveBinaryTargets(generator);
    }
    return Promise.resolve(data);
  }
  const error = (0, import_chunk_XKZ6CBLA.z)(configEither.left).with({ type: "wasm-error" }, (e) => {
    debugErrorType(e);
    if ((0, import_chunk_LMVSIVKQ.isWasmPanic)(e.error)) {
      const { message, stack } = (0, import_chunk_LMVSIVKQ.getWasmError)(e.error);
      const panic = new import_chunk_LMVSIVKQ.RustPanic(
        /* message */
        message,
        /* rustStack */
        stack,
        /* request */
        "@prisma/prisma-schema-wasm get_config",
        "FMT_CLI"
        /* FMT_CLI */
      );
      return panic;
    }
    const errorOutput = e.error.message;
    return new GetConfigError((0, import_chunk_WSZXPLJQ.parseQueryEngineError)({ errorOutput, reason: e.reason }));
  }).with({ type: "validation-error" }, (e) => {
    return new GetConfigError({
      _tag: "parsed",
      errorCode: SCHEMA_VALIDATION_ERROR_CODE,
      reason: (0, import_chunk_WSZXPLJQ.createSchemaValidationError)(e.reason),
      message: formatErrors(e.error)
    });
  }).otherwise((e) => {
    debugErrorType(e);
    return new GetConfigError({ _tag: "unparsed", message: e.error.message, reason: e.reason });
  });
  throw error;
}
async function resolveBinaryTargets(generator) {
  for (const binaryTarget of generator.binaryTargets) {
    if (binaryTarget.fromEnvVar && process.env[binaryTarget.fromEnvVar]) {
      const value = JSON.parse(process.env[binaryTarget.fromEnvVar]);
      if (Array.isArray(value)) {
        generator.binaryTargets = value.map((v) => ({ fromEnvVar: null, value: v }));
        await resolveBinaryTargets(generator);
      } else {
        binaryTarget.value = value;
      }
    }
    if (binaryTarget.value === "native") {
      binaryTarget.value = await (0, import_get_platform.getBinaryTargetForCurrentPlatform)();
      binaryTarget.native = true;
    }
  }
  if (generator.binaryTargets.length === 0) {
    generator.binaryTargets = [{ fromEnvVar: null, value: await (0, import_get_platform.getBinaryTargetForCurrentPlatform)(), native: true }];
  }
}
function formatErrors(errors) {
  const formattedErrors = errors.map((e) => (0, import_chunk_C47SCASR.relativizePathInPSLError)(e.message)).join("\n\n");
  const errorCount = `Validation Error Count: ${errors.length}`;
  return `${formattedErrors}
${errorCount}`;
}
