// @prisma/internals/dist/chunk-5QZLAN5R.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var chunk_5QZLAN5R_exports = {};
__export(chunk_5QZLAN5R_exports, {
  MergeSchemasError: () => MergeSchemasError,
  mergeSchemas: () => mergeSchemas
});
module.exports = __toCommonJS(chunk_5QZLAN5R_exports);
var import_chunk_WSZXPLJQ = require("./chunk-WSZXPLJQ.js");
var import_chunk_R6QH57HZ = require("./chunk-R6QH57HZ.js");
var import_chunk_C47SCASR = require("./chunk-C47SCASR.js");
var import_chunk_6Y7FKZ6B = require("./chunk-6Y7FKZ6B.js");
var import_chunk_SMJRP5CD = require("./chunk-SMJRP5CD.js");
var import_chunk_5DBOS77Y = require("./chunk-5DBOS77Y.js");
var import_chunk_XKZ6CBLA = require("./chunk-XKZ6CBLA.js");
var import_chunk_LMVSIVKQ = require("./chunk-LMVSIVKQ.js");
var import_chunk_PG5FDKSF = require("./chunk-PG5FDKSF.js");
var import_chunk_4VNS5WPM = require("./chunk-4VNS5WPM.js");
var import_debug = __toESM(require("@prisma/debug"));
var import_function = (0, import_chunk_4VNS5WPM.__toESM)((0, import_chunk_R6QH57HZ.require_function)());
var debug = (0, import_debug.default)("prisma:mergeSchemas");
var MergeSchemasError = class extends Error {
  constructor(params) {
    const constructedErrorMessage = (0, import_chunk_XKZ6CBLA.z)(params).with({ _tag: "parsed" }, ({ errorCode, message, reason }) => {
      const errorCodeMessage = errorCode ? `Error code: ${errorCode}` : "";
      return `${reason}
${errorCodeMessage}
${(0, import_chunk_C47SCASR.relativizePathInPSLError)(message)}`;
    }).with({ _tag: "unparsed" }, ({ message, reason }) => {
      const detailsHeader = (0, import_chunk_PG5FDKSF.red)((0, import_chunk_PG5FDKSF.bold)("Details:"));
      return `${reason}
${detailsHeader} ${message}`;
    }).exhaustive();
    const errorMessageWithContext = `${constructedErrorMessage}
[Context: mergeSchemas]`;
    super((0, import_chunk_6Y7FKZ6B.addVersionDetailsToErrorMessage)(errorMessageWithContext));
    this.name = "MergeSchemasError";
  }
};
function mergeSchemas(options) {
  const debugErrorType = (0, import_chunk_WSZXPLJQ.createDebugErrorType)(debug, "mergeSchemasWasm");
  debug(`Using mergeSchemas Wasm`);
  const mergeSchemasEither = (0, import_function.pipe)(
    (0, import_chunk_5DBOS77Y.tryCatch)(
      () => {
        const params = JSON.stringify({
          schema: options.schemas
        });
        return import_chunk_SMJRP5CD.prismaSchemaWasm.merge_schemas(params);
      },
      (e) => ({
        type: "wasm-error",
        reason: "(mergeSchemas wasm)",
        error: e
      })
    )
  );
  if ((0, import_chunk_5DBOS77Y.isRight)(mergeSchemasEither)) {
    return mergeSchemasEither.right;
  }
  const error = (0, import_chunk_XKZ6CBLA.z)(mergeSchemasEither.left).with({ type: "wasm-error" }, (e) => {
    debugErrorType(e);
    console.error("");
    if ((0, import_chunk_LMVSIVKQ.isWasmPanic)(e.error)) {
      const { message, stack } = (0, import_chunk_LMVSIVKQ.getWasmError)(e.error);
      debug(`Error merging schemas: ${message}`);
      debug(stack);
      const panic = new import_chunk_LMVSIVKQ.RustPanic(
        /* message */
        message,
        /* rustStack */
        stack,
        /* request */
        "@prisma/prisma-schema-wasm merge_schemas",
        "FMT_CLI"
        /* FMT_CLI */
      );
      return panic;
    }
    const errorOutput = e.error.message;
    return new MergeSchemasError((0, import_chunk_WSZXPLJQ.parseQueryEngineError)({ errorOutput, reason: e.reason }));
  }).exhaustive();
  throw error;
}
